Port to 1.17

pupnewfster opened the pull request.

pupnewfster committed:
Remove deprecated methods that have replacements and make the following minor fixes:
- FluidStackHelper, TestIngredientHelper, and DebugIngredientHelper getMatch ignores the unique identifier based on context (only really mattered for FluidStackHelper)
- FluidStackHelper missing getWildcardId override

pupnewfster committed:
Initial port to 1.17.1 and mojang class names that are now needed. Also for now added a game version to the curseforge block as it was failing to evaluate. Compiles but various things currently render incorrectly

shartte committed:
Several 1.17 rendering fixes (#1)

pupnewfster committed:
Manually specify automatic module name as workaround for FG not being able to calculate it properly for mapped deps as JEI is heavily used in mod dev environments

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/mezz/JustEnoughItems?pullRequest=2426) <br/>All committers have signed the CLA.

pupnewfster committed:
Disable fast item stack renderer for now as it has a bunch of lighting issues, and remove some TODOs related to things that don't actually seem to be needed anymore and update forge

mezz committed:
modernize mezz.jei.util.AnnotatedInstanceUtil#getInstances

mezz committed:
update uses of deprecated fields

mezz committed:
avoid using RenderSystem.applyModelViewMatrix() for ItemStackRenderer

mezz committed:
remove unused imports

pupnewfster committed:
Update forge and use new event to add the client side reload listener at the proper time. Also update the loader version in the mods.toml

pupnewfster committed:
Fix debug recipe category rendering oddly at times due to the font drawing changing shader state in a way that ExtendedButton is unable to handle. Also make DebugRecipeCategory support drawing the ingredient name of items hovered in bookmarks

pupnewfster committed:
Fix ItemStackFastRenderer and batch rendering

mezz committed:
Fix RecipeGuiLogic#setCategoryFocus

mezz committed:
seems fine to override the new init

It has basically the same functionality.
If anyone canceled the event, I hope they know what they're doing ...

pupnewfster committed:
Adjust how we listen to tag events in preparation for when https://github.com/MinecraftForge/MinecraftForge/pull/7898 gets merged to not break as hard.

mezz committed:
improve logic for RecipesGui adding recipe transfer butons

pupnewfster committed:
Update forge version

pupnewfster committed:
Giving still works just fine and updates properly on client

pupnewfster committed:
Fix some javadoc issues from class renames

pupnewfster committed:
Remove some missed unused imports and update forge and account for a breaking rename

pupnewfster committed:
Adjust version number and remove comment related to burn time

mezz commented on src/api/java/mezz/jei/api/constants/VanillaRecipeCategoryUid.java:
```diff
@@ -33,35 +35,35 @@
 	 * The campfire furnace recipe category.
 	 *
-	 * Automatically includes every {@link FurnaceRecipe}
+	 * Automatically includes every {@link SmeltingRecipe}
```
should this be `CampfireCookingRecipe`?

pupnewfster replied:
Yes, I misread the line when I was changing it and thought it was some generic abstract/anonymous implementation mojang was using so didn't rename it. Upon re-evaluation it does seem that is a proper class.

mezz commented on src/main/java/mezz/jei/plugins/vanilla/ingredients/item/ItemStackListFactory.java:
```diff
@@ -46,6 +46,7 @@
 	}
 
 	private void addItemStack(StackHelper stackHelper, ItemStack stack, List<ItemStack> itemList, Set<String> itemNameSet) {
+		//TODO: Test to make sure this is actually fixed in 1.17 and if so remove this check
```
good todo 👍

pupnewfster committed:
Fix doc
